src/main/java/com/jfrog/xray/client/impl/services/graph/GraphImpl.java

```diff
@@ -0,0 +1,95 @@
+    private static final int MAX_ATTEMPTS = 60;
+    private static final int SYNC_SLEEP_INTERVAL = 5000;
+    private final XrayClient xray;
+
+    public GraphImpl(XrayClient xray) {
+        this.xray = xray;
+        mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+        FilterProvider filters = new SimpleFilterProvider().setFailOnUnknownId(false)
+                .addFilter("xray-graph-filter", SimpleBeanPropertyFilter.filterOutAllExcept("component_id", "nodes"));
+        mapper.setFilterProvider(filters);
+    }
+
+    @Override
+    public GraphResponse graph(DependencyTree dependencies, String projectName) throws IOException, InterruptedException {
+        if (projectName == null || projectName.isEmpty()) {
+            return graph(dependencies);
+        }
+        if (dependencies == null) {
+            return new GraphResponseImpl();
+        }
+        return this.post("?project=" + projectName, dependencies);
+    }
+
+    @Override
+    public GraphResponse graph(DependencyTree dependencies) throws IOException, InterruptedException {
+        if (dependencies == null) {
+            return new GraphResponseImpl();
+        }
+        return this.post("", dependencies);
+    }
+
+    private GraphResponse post(String params, Object body) throws IOException, InterruptedException {
+        HttpEntity entity = null;
+        // First, request a scan from Xray.
+        try (CloseableHttpResponse response = xray.post("scan/graph" + params, body, mapper)) {
+            entity = response.getEntity();
+            GraphResponse requestResponse = mapper.readValue(response.getEntity().getContent(), GraphResponseImpl.class);
+            String scanId = requestResponse.getScanId();
+            // If no context was provided (project name), we would like to receive all known vulnerabilities.
+            String includeVulnerabilities = !params.isEmpty() ? "" : "&include_vulnerabilities=true";
+            // Xray will respond with 201 until the completion of the scan, then 200 will be returned.
+            for (int i = 0; i < MAX_ATTEMPTS; i++) {
+                try (CloseableHttpResponse res = xray.get("scan/graph/" + scanId + "?include_licenses=true" + includeVulnerabilities)) {
+                    StatusLine statusLine = res.getStatusLine();
+                    int statusCode = statusLine.getStatusCode();
+                    if (statusCode == HttpStatus.SC_OK) {
+                        // We got an answer (scan completed), return it.
+                        entity = res.getEntity();
+                        return mapper.readValue(res.getEntity().getContent(), GraphResponseImpl.class);
+                    }
+                } finally {
+                    EntityUtils.consumeQuietly(entity);
+                }
+                // Wait between polling attempts.
+                Thread.sleep(SYNC_SLEEP_INTERVAL);
+            }
+        } finally {
+            EntityUtils.consumeQuietly(entity);
```

Comment:
This seems to be redundant, because the entity is already consumed in line 81 above.

Reply:
But it is not the same entity, Do you prefer me to change the variable name?